Code/GraphMol/CIPLabeler/catch_tests.cpp

```diff
@@ -554,3 +556,128 @@ TEST_CASE("GitHub Issue #5142", "[bug][accurateCIP]") {
+    5.8522   -9.1147    0.4514 H   0  0  0  0  0  0  0  0  0  0  0  0
+    7.1601   -8.2997    0.2353 C   0  0  2  0  0  0  0  0  0  0  0  0
+    7.5774   -7.5989    1.5700 C   0  0  1  0  0  0  0  0  0  0  0  0
+   10.6577   -4.0099    0.1683 H   0  0  0  0  0  0  0  0  0  0  0  0
+   11.0808   -6.4894   -0.1982 C   0  0  2  0  0  0  0  0  0  0  0  0
+    8.9123   -5.4699   -0.9424 C   0  0  1  0  0  0  0  0  0  0  0  0
+    8.9155   -8.2329    2.0184 C   0  0  2  0  0  0  0  0  0  0  0  0
+   11.0766   -7.5513    0.9147 C   0  0  1  0  0  0  0  0  0  0  0  0
+   10.6780   -6.9857    3.3021 H   0  0  0  0  0  0  0  0  0  0  0  0
+    7.5861   -5.8939   -0.2667 C   0  0  2  0  0  0  0  0  0  0  0  0
+    6.9973   -5.0597    1.9875 H   0  0  0  0  0  0  0  0  0  0  0  0
+    9.3193   -5.8429    1.5048 C   0  0  2  0  0  0  0  0  0  0  0  0
+    9.4740   -4.7173    2.5743 H   0  0  0  0  0  0  0  0  0  0  0  0
+    9.9809   -7.1457    1.9811 C   0  0  1  0  0  0  0  0  0  0  0  0
+    7.8270   -6.1245    1.2483 C   0  0  2  0  0  0  0  0  0  0  0  0
+   10.0023   -5.4308    0.1366 C   0  0  1  0  0  0  0  0  0  0  0  0
+    8.9125   -8.9385   -1.4737 C   0  0  2  0  0  0  0  0  0  0  0  0
+    8.2374   -9.3522   -0.1173 C   0  0  1  0  0  0  0  0  0  0  0  0
+   10.4116   -8.6461   -1.2148 C   0  0  1  0  0  0  0  0  0  0  0  0
+    8.7726   -9.9758   -2.5047 H   0  0  0  0  0  0  0  0  0  0  0  0
+    8.2577   -7.6250   -1.9478 C   0  0  2  0  0  0  0  0  0  0  0  0
+  1 40  1  0  0  0  0
+  2  3  1  0  0  0  0
+  3  9  1  0  0  0  0
+  3 25  1  0  0  0  0
+  3 40  1  0  0  0  0
+  4 25  1  0  0  0  0
+  5  7  1  0  0  0  0
+  6 29  1  0  0  0  0
+  7 21  1  0  0  0  0
+  7 29  1  0  0  0  0
+  7 40  1  0  0  0  0
+  8  9  1  0  0  0  0
+  9 24  1  0  0  0  0
+  9 38  1  0  0  0  0
+ 10 38  1  0  0  0  0
+ 11 12  1  0  0  0  0
+ 12 17  1  0  0  0  0
+ 12 27  1  0  0  0  0
+ 12 38  1  0  0  0  0
+ 13 24  1  0  0  0  0
+ 14 27  1  0  0  0  0
+ 15 26  1  0  0  0  0
+ 16 17  1  0  0  0  0
+ 17 26  1  0  0  0  0
+ 17 37  1  0  0  0  0
+ 18 37  1  0  0  0  0
+ 19 22  1  0  0  0  0
+ 20 21  1  0  0  0  0
+ 21 22  1  0  0  0  0
+ 21 37  1  0  0  0  0
+ 22 26  1  0  0  0  0
+ 22 34  1  0  0  0  0
+ 23 35  1  0  0  0  0
+ 24 27  1  0  0  0  0
+ 24 35  1  0  0  0  0
+ 25 29  1  0  0  0  0
+ 25 35  1  0  0  0  0
+ 26 33  1  0  0  0  0
+ 27 33  1  0  0  0  0
+ 28 33  1  0  0  0  0
+ 29 34  1  0  0  0  0
+ 30 34  1  0  0  0  0
+ 31 32  1  0  0  0  0
+ 31 33  1  0  0  0  0
+ 31 35  1  0  0  0  0
+ 31 34  1  0  0  0  0
+ 36 37  1  0  0  0  0
+ 36 38  1  0  0  0  0
+ 36 39  1  0  0  0  0
+ 36 40  1  0  0  0  0
+M  END
+
+  )";
+  
+  auto mol = std::unique_ptr<RDKit::RWMol>(MolBlockToMol(molBlock,true, false));
+  REQUIRE(mol);
+  bool caughtIt = false;
+  try {
+    CIPLabeler::assignCIPLabels(*mol, 100000);
+  }
+  catch(const CIPLabeler::MaxIterationsExceeded &e)
+  {
+    caughtIt = true;
+  } 
+  CHECK(caughtIt);
```

Comment:
catch2 makes this pattern super easy:
```suggestion
  CHECK_THROWS_AS(CIPLabeler::assignCIPLabels(*mol, 100000), CIPLabeler::MaxIterationsExceeded);
```